addons/web/controllers/export.py

```diff
@@ -559,8 +560,11 @@ def base(self, data):
         else:
             columns_headers = [val['label'].strip() for val in fields]
 
+        records = Model.browse(ids) if ids else Model.search(domain)
```

Comment:
Because you push this line before the condition for groupby export, the complexity done with `id:array_agg` + search for reordering inside `insert_leaf` became obsolete (and inefficient). (see previous comment).